dashboard/app/entities.go

```diff
@@ -123,6 +123,21 @@ type Bug struct {
 	Labels         []BugLabel
 	DiscussionInfo []BugDiscussionInfo
 	TreeTests      BugTreeTestInfo
+	FixCandidates  []BugFixCandidate
+}
+
+type FixCandidateSource string
+
+const (
+	BisectionFixCandidate FixCandidateSource = "bisection"
+)
+
+type BugFixCandidate struct {
```

Comment:
How do we plan to use these?

If we plan to report these somehow, then it may need some reporting state as well.

I am thinking if some of these uses are also applicable/equal for normal fix bisections and if we should create BugFixCandidate for normal fix bisections. E.g. if we found a fix candidate via cross-tree bisection, we probably also want to send a notification to the bug saying "this is a potential fix".

Retrospectively adding Time for all events is useful and allows to infer statistics on latencies/correlations later. But the Job already contains Time. But also this whole struct is merely a copy of info from the Job. Do we need it at all? Maybe a single field saying there is a fix job like we have for other bisections?

Reply:
> If we plan to report these somehow, then it may need some reporting state as well.

Yes, the idea was to
a) Report fix candidates to bug discussions.
b) Display some unified view of missing backports for every namespace (or, rather, for every tree?). That should be most useful for people responsible for fix backporting.

> I am thinking if some of these uses are also applicable/equal for normal fix bisections and if we should create BugFixCandidate for normal fix bisections.

I also thought about this, though it seems to be a different kind of thing. Normal fix bisection only affects whether the bug should remain open on the syzbot dashboard. Cross tree bisection, if it's correct, needs to be followed by an actual backport to the fuzzed tree -- i.e. it's not about finding some already landed fix, but only about syzbot proposing a kernel code change.

The message could go like `syzbot has performed a fix bisection on the mainline tree and identified the missing commit that could resolve the reported problem`.

> Do we need it at all? Maybe a single field saying there is a fix job like we have for other bisections?

An interesting question is how likely it is that there'll appear other good sources of potential fixes. I thought about looking at similar bugs and re-testing commits from `#syz fix` (at least run the repro on `commit` and `commit^1`), but it seemed quite unreliable when I ran the experiments manually. Maybe if/when we learn to better relate bugs from different namespaces, then it can become more useful.

If cross-tree bisection jobs are to stay the only source of potential fix candidates, we could indeed just store some flag in the `Bug` structure. That should be enough for bug report generation and for the `/bug?` page generation. The bug discussion could be notified via the existing job reporting mechanism then. 

The overall per-namespace view would be somewhat problematic in this case though as we'd have to go to each open bug with a fix candidate, query its corresponding cross-tree job, then query its corresponding build object (to figure out the tree).

Reply:
Overall it's better complicate things and add entities when they are needed rather than proactively. This also has good chances of staying the only source of fix commits forever. There are lots of potential complications, e.g. what if fuzzing is not the only source of bugs (e.g. we do CI unit-testing)?
If we relate bugs to find fixing commits that will probalby also be a Job, so potentially we can handle then uniformly as Jobs of different subtypes.

Reply:
>The overall per-namespace view would be somewhat problematic in this case though as we'd have to go to each open bug with a fix candidate, query its corresponding cross-tree job, then query its corresponding build object (to figure out the tree).

I am not sure we need to split them per-tree, but Job should have the tree.
Fetching bugs and jobs for this page looks reasonable.

Reply:
I've updated the code not to duplicate the info in `Bug`.